airflow/contrib/hooks/aws_hook.py

```diff
@@ -111,6 +111,8 @@ def _get_credentials(self, region_name):
                         'aws_access_key_id']
                     aws_secret_access_key = extra_config[
                         'aws_secret_access_key']
+                    # If there is an external session token use this token
+                    aws_session_token = extra_config['aws_session_token']
```

Comment:
This is going to throw a KeyError is the extra json doesn't contain `aws_session_token`.

Additionally: The aws_session_token being in the extra or not should not be dependent on aws_access_key_id/aws_secret_access_Key being in the extra dict as the "better" place to put those is the login/pass field. (We should deprecated reading these from the extra dict in fact. Separate PR for that though.) 

Please expand the tests to cover more cases (where we have no session token, where we have session token but no key/secret in extra)

Reply:
okay I agree that we should check because of the exception. 
Also then I would move it to the very bottom of the if statement and check if the session token is empty before.
Also extended test and ocs